backend/app/services/prompt_draft_service.py

```
@@ -0,0 +1,558 @@
+        if isinstance(text, str):
+            normalized = text.strip()
+            if normalized:
+                texts.append(normalized)
+    return texts
+
+
+def _parse_tool_arguments(raw_arguments: Any) -> dict[str, Any]:
+    if not isinstance(raw_arguments, str) or not raw_arguments.strip():
+        return {}
+    try:
+        parsed = json.loads(raw_arguments)
```

Ruff (0.15.7)

[error] Undefined name `json`

(F821)